src/compact_live.rs

```diff
@@ -0,0 +1,449 @@
+    const DEM_KEY_LENGTH: usize,
+    KmacKey: SymKey<KMAC_KEY_LENGTH>,
+    DemScheme: Dem<DEM_KEY_LENGTH>,
+    CustomError: std::error::Error + CallbackError,
+>:
+    FindexCallbacks<CustomError, UID_LENGTH>
+    // to get `batch_fetch_chains()`
+    + FindexCompact<
+        UID_LENGTH,
+        BLOCK_LENGTH,
+        CHAIN_TABLE_WIDTH,
+        MASTER_KEY_LENGTH,
+        KWI_LENGTH,
+        KMAC_KEY_LENGTH,
+        DEM_KEY_LENGTH,
+        KmacKey,
+        DemScheme,
+        CustomError,
+    >
+{
+    /// Ratio of the noise over the real data.
+    const NOISE_RATIO: f64;
+
+    /// Number of noisy Entry Table UIDs to compact in a row.
+    const BATCH_SIZE: usize;
+
+    /// Selects a set of Entry Table UIDs to reindex (target set) and a set of Entry Table UIDs
+    /// used as noise (noise set).
+    ///
+    /// These sets are mutually exclusive and the ratio of the cardinal of the noise set over the
+    /// target set is given by `Self::NOISE_RATIO`.
+    ///
+    /// Returns the noisy taget set (target set + noise set) and the noise set.
+    /// A HashSet is used for the noise set because lookup should be fast.
+    ///
+    /// # Parameters
+    ///
+    /// - `rng`                             : secure random number generator
+    /// - `num_reindexing_before_full_set`  : number of compact operations need to compact all the
+    /// base
+    async fn select_noisy_uids(
+        &self,
+        rng: &mut impl CryptoRngCore,
+        num_reindexing_before_full_set: u32
+    ) -> Result<(Vec<Uid<UID_LENGTH>>, HashSet<Uid<UID_LENGTH>>), Error<CustomError>> {
+        if Self::NOISE_RATIO > 1f64 {
+            return Err(Error::CryptoError(format!(
+                "noise ration cannot be greater than 1 ({} given)",
+                Self::NOISE_RATIO
+            )));
+        }
+
+        let uids = self.fetch_all_entry_table_uids().await?;
+
+        // TODO (TBZ): find which probability is associated to this formula.
+        let entry_table_length = uids.len() as f64;
+        let n_compact = ((entry_table_length * (entry_table_length.log2() + 0.58))
+            / f64::from(num_reindexing_before_full_set))
+        .ceil() as usize;
+
+        // The number of compacted UIDs should leave enough unused UIDs for the noise.
+        if (n_compact as f64  * (1f64 + Self::NOISE_RATIO)) > uids.len() as f64 {
+            return Err(Error::CryptoError(format!(
+                "Number of Entry Table UIDs to compact ({n_compact}) should not be greater than {}",
+                uids.len() as f64 / (1f64 + Self::NOISE_RATIO)
+            )));
+        }
+
+        // Upper bounds are used for capacity in order to avoid reallocations:
+        // - the number of noise UIDs is lower than `n_compact`;
+        let mut noise = HashSet::with_capacity(n_compact);
+        // - the total number of UIDs (target set + noise set) is lower than `2*n_compact`.
+        let mut noisy_uids_to_compact = Vec::with_capacity(2 * n_compact );
```

Comment:
Trailing whitespace within parenthesis

Reply:
Yes. I don't know why, it seems Rustfmt does not work anymore on this file on my machine...